glambie/data/timeseries.py

```diff
@@ -390,6 +382,54 @@ def convert_timeseries_to_unit_gt(self, include_area_change: bool = True,
             raise NotImplementedError(
                 "Conversion to Gt not implemented yet for Timeseries with unit '{}'".format(self.unit))
 
+    def apply_area_change(self, rgi_area_version: int = 6, apply_change: bool = True) -> Timeseries:
+        """
+        Applies or removes a changing area to observed changes.
+        Returns a copy of itself with the converted timeseries.
+
+        Parameters
+        ----------
+        rgi_area_version : int, optional
+            version of RGI used for area change, by default 6
+        apply_change : bool, optional
+            Describes if the area change should be applied or removed
+            If set to False, the area change is removed rather than applied, by default True
+
+        Returns
+        -------
+        Timeseries
+            A copy of the Timeseries object containing the converted timeseries data.
+
+        """
+        if self.unit not in ["mwe", "m"]:
+            raise AssertionError("Area change should only applied to 'm' or 'mwe'.")
+        # get area
+        if rgi_area_version == 6:
+            glacier_area = self.region.rgi6_area
+        elif rgi_area_version == 7:
+            glacier_area = self.region.rgi7_area
+
+        object_copy = self.copy()
+        # conversion with area change
+        t_0 = self.region.area_change_reference_year
+        area_change = self.region.area_change
+        adjusted_changes = []
+        adjusted_areas = []
+        for _, row in self.data.as_dataframe().iterrows():
+            t_i = (row["start_dates"] + row["end_dates"]) / 2
+            adjusted_area = glacier_area + (t_i - t_0) * (area_change / 100) * glacier_area
+            if apply_change:
+                adjusted_changes.append(glacier_area / adjusted_area * row.changes)
+            else:  # remove change
+                adjusted_changes.append(row.changes / (glacier_area / adjusted_area))
+            adjusted_areas.append(adjusted_area)
+        # @TODO: add uncertainty ?
```

Comment:
One thing on this - I typically avoid iterrows() like the plague due to its performance issues: https://stackoverflow.com/questions/24870953/does-pandas-iterrows-have-performance-issues

So I'd typically write this as a list comprehension:

```
df = self.data.as_dataframe()
adjusted_changes = [compute_adjusted_change(start_dates, end_dates, changes)
                    for start_dates, end_dates, changes
                    in zip(df["start_dates"], df["end_dates"], df["changes"])]
```
with compute_adjusted_change() as a little internal method. This could probably be used without retrieving the data as a dataframe in the first place.

There's others ways too: https://towardsdatascience.com/heres-the-most-efficient-way-to-iterate-through-your-pandas-dataframe-4dad88ac92ee.

Of course this does not matter as much if the underlying data isn't that big - I don't quite have the full picture here!

Reply:
Thanks Julia,
It's only really small datasets, so performance is not really an issue here.
But this is good to know that iterrows is really slow and I changed it to list comprehension here instead.

Reply:
Good stuff!